pkg/server/plugin/nodeattestor/azuremsi/msi.go

```diff
@@ -277,26 +272,34 @@ func (p *MSIAttestorPlugin) Configure(_ context.Context, req *configv1.Configure
 			if err != nil {
 				return nil, status.Errorf(codes.Internal, "unable to create client for tenant %q: %v", tenantID, err)
 			}
+
 		case tenant.UseMSI:
+			p.log.Warn("use_msi is deprecated and will be removed in a future release")
+			fallthrough // use default credential which attempts to fetch credentials using MSI
+
+		default:
 			instanceMetadata, err := p.hooks.fetchInstanceMetadata(http.DefaultClient)
 			if err != nil {
 				return nil, err
 			}
-			cred, err := p.hooks.msiCredential()
+			cred, err := azidentity.NewDefaultAzureCredential(
```

Comment:
can you add a test case for this failing?

Reply:
With credential chaining happening in the SDK, errors are ignored. We would need to wrap this call behind a custom hook to test this scenario